### PR TITLE
[FEAT]: Display the number of Completed Question on Each Section

### PR DESCRIPTION
**Pull request by root-daemon**
#13 
- Displays the number of Completed Questions, Opened Questions and Unopened questions right next the levels
- Created a recursive function calculateQuestionsStatus to calculate the status of the questions
- Fixed the styling to make sure it doesn't break the layout

### CODE REVIEW

**Comment by Rahuletto** [comment]
Can I get the screenshot of what it looks like?

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,6 @@ yarn-error.log*
 # typescript
 *.tsbuildinfo
 next-env.d.ts
+
+#middleware
```
Comment (Rahuletto) [inline .gitignore:38]: Remove these lines, it's useless

File: components/Elements/Dialog/QuestionsProgress.tsx [hunk 2]
```diff
@@ -38,7 +38,17 @@ const QuestionsProgress = ({
   }, [user]);
 
   const [index, setIndex] = useState(0);
+  function calculateQuestionStatus(node: any, status: number): number {
+    console.log(node);
```
Comment (Rahuletto) [inline components/Elements/Dialog/QuestionsProgress.tsx:42]: Remove debugging elements in code

File: pages/course.tsx [hunk 3]
```diff
@@ -6,14 +6,6 @@ import dynamic from 'next/dynamic';
...
 import { useEffect, useState } from 'react';
 
-const SettingsButton = dynamic(
-  () =>
-    import('@/components/Elements/Buttons/SettingsButton').then(
-      (mod) => mod.default
-    ),
-  { ssr: false }
-);
-
```
Comment (Rahuletto) [inline pages/course.tsx:16]: Why do u have to remove these lines?

**Review by Rahuletto — CHANGES_REQUESTED** [review]

File: pages/course.tsx (hunk 3 above)
Comment (root-daemon) [reply pages/course.tsx:16]: It wasn't used, unused import.

**Comment by Rahuletto** [comment]
Screenshot?

File: components/Elements/Dialog/QuestionsProgress.tsx [hunk 4]
```diff
@@ -58,7 +57,7 @@ const QuestionsProgress = ({
       setIndex(index - 1);
     }
   }
-
+  ``;
```
Comment (Rahuletto) [inline components/Elements/Dialog/QuestionsProgress.tsx:60]: ?

**Review by Rahuletto — CHANGES_REQUESTED** [review]

**Review by Rahuletto — APPROVED** [review]

**Review by Rahuletto — APPROVED** [review]

**Pull request merged**